File: src/Shared/Logistics.Shared.Models/Subscription/SubscriptionDto.cs
using Logistics.Domain.Primitives.Enums;

namespace Logistics.Shared.Models;

public record SubscriptionDto
{
    public Guid? Id { get; set; }
    public SubscriptionStatus Status { get; set; }
    public TenantDto? Tenant { get; set; }
    public SubscriptionPlanDto? Plan { get; set; }
    public string? StripeSubscriptionId { get; set; }
    public string? StripeCustomerId { get; set; }
}

File: src/Client/Logistics.OfficeApp/src/app/core/api/models/subscription/subscription.dto.ts
import {SubscriptionStatus} from "./enums";
import {SubscriptionPlanDto} from "./subscription-plan.dto";

export interface SubscriptionDto {
  id: string;
  status: SubscriptionStatus;
  plan?: SubscriptionPlanDto;
  startDate: Date;
  endDate?: Date;
  nextBillingDate?: Date;
  trialEndDate?: Date;
  stripeSubscriptionId?: string;
  stripeCustomerId?: string;
}
